app/helpers/feature_flag_helper.rb

```diff
@@ -1,89 +1,61 @@
-# A module intended to be used as a mixin with controllers whose
-# actions / views want to access information about feature flags.
+# These methods are intended to be called statically
+# Call init first to initialize the state for the request
 module FeatureFlagHelper
 
   class FeatureFlagNotEnabledError < StandardError; end
 
-  def self.included(target)
-    target.include InstanceMethods
-    target.extend ClassMethods
-  end
-
-  module InstanceMethods
-    def feature_enabled?(feature_name)
-      feature_flags.include? feature_name
-    end
-
-    def assert_feature_enabled(feature_name)
-      raise FeatureFlagNotEnabledError.new("Missing required feature: #{feature_name}") unless feature_enabled?(feature_name)
-    end
+  module_function
 
-    def with_feature(feature_name, &block)
-      block.call if feature_enabled?(feature_name)
-    end
+  def init(request, is_admin)
+    RequestStore.store[:feature_flags] ||= fetch_feature_flags(request, is_admin)
+  end
 
-    def feature_flags
-      RequestStore.store[:feature_flags] ||= fetch_feature_flags
-    end
+  def feature_enabled?(feature_name)
+    feature_flags.include? feature_name
+  end
 
-    def fetch_feature_flags
-      flags_from_service = FeatureFlagService::API::Api.features.get(community_id: community_id(request)).maybe[:features].or_else(Set.new)
+  def assert_feature_enabled(feature_name)
+    raise FeatureFlagNotEnabledError.new("Missing required feature: #{feature_name}") unless feature_enabled?(feature_name)
+  end
 
-      is_admin = called_with_admin_user?
-      temp_flags = fetch_temp_flags(is_admin, request.params, request.session)
+  def with_feature(feature_name, &block)
+    block.call if feature_enabled?(feature_name)
+  end
 
-      request.session[:feature_flags] = temp_flags
+  def feature_flags
+    RequestStore.store[:feature_flags]
```

Comment:
This should maybe raise if not initialized? The two step flow this now needs is more prone to user errors so providing a bit of help by clearly telling when the poor developer made a mistake would be nice.